QUARKUS-2736.md

```diff
@@ -0,0 +1,145 @@
+- we have necessary coverage on [MicroProfile 6.0 migration](https://github.com/quarkusio/quarkus/issues/31084) that is
+  based on Jakarta EE 10 Core Profile, to make sure there is no transient impact
+- ensure good user experience and simplicity of migration to Hibernate ORM 6.2:
+  - add coverage for [configuration property to work with database schemas generated for Hibernate ORM 5.6](https://github.com/quarkusio/quarkus/pull/31540)
+    as portion of these changes is related to Jakarta specification
+  - add coverage for `quarkus.datasource.db-version` [configuration property](https://github.com/quarkusio/quarkus/pull/31701)
+  - Hibernate ORM is lenient on configuration properties prefixed with `javax.persistence`, however all features are not
+    guaranteed to work with `javax`, therefore we should [encourage in upstream issue](https://github.com/quarkusio/quarkus/issues/31130)
+    to log warning or fail. This way, users are going to be warned early.
+- our test suites uses terminology [compliant to the specification](https://github.com/quarkusio/quarkus/pull/31700/files), for example:
+  - replace `JPA` with `Jakarta Persistence`
+  - replace `JAX-RS` with `Jakarta REST implementation` or `Jakarta REST`
+
+### Existing tests
+
+Naming migration of packages and imports from `javax` to `jakarta` has already been done, there was no
+regression caused by the migration, however we had to disable temporarily several tests using Quarkiverse extensions that don't have
+Jakarta version yet. Such examples are Quarkus Artemis, Kogito, Quarkus Pact and Quarkus Camel. Similarly, we disabled
+tests using external applications like Quarkus Quickstarts, Quarkus TODO, Quarkus Workshop etc. Enabling this test will
+sufficiently prove there is no regression.
+
+ArC changes are verified by [newly enabled ArC TCK tests](https://github.com/quarkusio/quarkus/pull/31589) that are run
+as part of the upstream CI.
+
+Our tests that use Hibernate 6.2 all [required changes](https://github.com/quarkus-qe/quarkus-test-suite/pull/1082) 
+regarding sequence generation, dialects and some also needed schema migration changes. Our Quarkus Jdk specifics Test Suite
+cover changes in Hibernate Query Language semantics not related to Jakarta as proved by [required changes](https://github.com/quarkus-qe/quarkus-jdkspecifics/pull/62).
+It's safe to say Quarkus Test Suite covers major Hibernate 6.2 breaking changes related to Jakarta EE 10 migration.
+
+Default exception mappers introduced by RESTEasy Classic are covered upstream as the mappers were actually disabled in
+reaction to failing tests.
+
+Elytron Security migration is sufficiently covered by Quarkus QE Test Suite, Quarkus Quickstarts run as part of upstream CI
+and upstream integration modules.
+
+## Impact on test suites and test environment
+
+Beefy Scenarios should cover newly introduced configuration properties that enforces legacy behavior and provides
```

Comment:
This should be adjusted to Quarkus QE TS, this will not be covered in Beefy TS.

Reply:
np, will do in a separate PR

Reply:
here you go https://github.com/quarkus-qe/quarkus-test-plans/pull/137